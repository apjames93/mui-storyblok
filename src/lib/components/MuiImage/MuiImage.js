import React from 'react';
import PropTypes from 'prop-types';
import StoryBlok from '../../utils/Storyblok';

const MuiImage = ({
  hrefLink,
  imageStyle,
  image,
  rootClass,
}) => {
  const styles = StoryBlok.anchorOrginToObj(rootClass);
  const imageStyles = StoryBlok.anchorOrginToObj(imageStyle);

  return (
    <>
      { hrefLink && (
      <a href={hrefLink} target="_self">
        <picture style={styles}>
          <img src={image} style={imageStyles} alt={image} />
        </picture>
      </a>
      )}
      { !hrefLink && (
        <picture style={styles}>
          <img src={image} style={imageStyles} alt={image} />
        </picture>
      )}
    </>
  );
};

export default MuiImage;

MuiImage.propTypes = {
  /**
   * Url for image to be displayed.
   */
  image: PropTypes.string.isRequired,
  /**
   * Url for image to be displayed.
   */
  hrefLink: PropTypes.string,
  /**
   * Css Styles for the image.
   */
  imageStyle: PropTypes.arrayOf(PropTypes.string),
  /**
   * Css Styles for the image.
   */
  rootClass: PropTypes.arrayOf(PropTypes.string),
};

MuiImage.defaultProps = {
  imageStyle: [],
  rootClass: [],
  hrefLink: null,
};
