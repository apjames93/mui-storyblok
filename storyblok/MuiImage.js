module.exports = {
  "name": "MuiImage",
  "display_name": null,
  "updated_at": "2020-06-11T21:51:54.532Z",
  "schema": {
    "hrefLink": {
      "type": "text",
      "default_value": "",
      "description": "Link to redirect on click"
    },
    "imageStyle": {
      "type": "options",
      "description": "Styling for the image",
      "min_options": "",
      "options": [
        {
          "value": "height: 25px",
          "name": "height"
        },
        {
          "value": "width: 250px",
          "name": "width"
        }
      ]
    },
    "image": {
      "type": "text",
      "description": "Url to the image",
      "regex": "",
      "default_value": "",
      "required": true
    },
    "rootClass": {
      "type": "options",
      "description": "Styling for the container of image\n",
      "options": [
        {
          "value": "position: absolute; right: 10",
          "name": "position right"
        }
      ]
    }
  },
  "image": null,
  "preview_field": null,
  "is_root": false,
  "preview_tmpl": null,
  "is_nestable": true,
  "all_presets": [],
  "preset_id": null,
  "real_name": "MuiImage",
  "component_group_uuid": null
}