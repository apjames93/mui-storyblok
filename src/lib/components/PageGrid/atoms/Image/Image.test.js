import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import MuiImage from './Image';

function setup(href = null) {
  const props = {
    hrefLink: href,
    image: 'www.applejuice.com',
  };

  const comp = mount(<MuiImage {...props} />);
  return { comp, props };
}

describe('<MuiImage />', () => {
  it('renders MuiImage', () => {
    const { comp } = setup();
    expect(comp).toBeDefined();
  });

  test('snapshot with MuiImage', () => {
    const { props } = setup('/home');

    const tree = renderer.create(<MuiImage {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
