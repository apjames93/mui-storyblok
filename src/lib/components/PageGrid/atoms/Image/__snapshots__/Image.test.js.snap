// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MuiImage /> snapshot with MuiImage 1`] = `
<div
  className=""
  data-blok-c=""
  data-blok-uid=""
>
  <picture
    style={Object {}}
  >
    <img
      alt="www.applejuice.com"
      src="www.applejuice.com"
      style={
        Object {
          "height": "500px",
          "width": "500px",
        }
      }
    />
  </picture>
</div>
`;
